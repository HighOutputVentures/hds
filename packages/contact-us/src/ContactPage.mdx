import { Canvas, Meta, Story, ArgsTable } from '@storybook/addon-docs';

import StorybookHeader from '../../../../components/StorybookHeader';

<Meta title="Layouts/Contact/Contact Page" />

<StorybookHeader
  title="HDS Contact Us"
  command="npm i @highoutput/hds-contact-us"
/>

<Canvas>
  <Story id="layouts-contact-contact-page--default"></Story>
</Canvas>

<ArgsTable />
