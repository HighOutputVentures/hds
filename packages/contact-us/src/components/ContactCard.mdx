import { Canvas, Meta, Story, ArgsTable } from '@storybook/addon-docs';

import StorybookHeader from '../../../../components/StorybookHeader';

<Meta title="Layouts/Contact/Contact Card" />

<StorybookHeader
  title="HDS Contact Us"
  command="npm i @highoutput/hds-contact-us"
/>

<Canvas>
  <Story id="layouts-contact-contact-card--default"></Story>
</Canvas>

<ArgsTable />
