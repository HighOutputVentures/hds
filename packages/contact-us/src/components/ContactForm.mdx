import { Canvas, Meta, Story, ArgsTable } from '@storybook/addon-docs';

import StorybookHeader from '../../../../components/StorybookHeader';

<Meta title="Layouts/Contact/Contact Form" />

<StorybookHeader
  title="HDS Contact Us"
  command="npm i @highoutput/hds-contact-us"
/>

<Canvas>
  <Story id="layouts-contact-contact-form--default"></Story>
</Canvas>

<ArgsTable />
