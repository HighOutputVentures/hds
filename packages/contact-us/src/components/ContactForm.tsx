import { Alert, AlertDescription, AlertIcon, Button } from '@chakra-ui/react';
import { Box, Stack } from '@highoutput/hds';
import { MultilineField, TextField } from '@highoutput/hds-forms';
import { yupResolver } from '@hookform/resolvers/yup';
import React, { FC } from 'react';
import { useForm } from 'react-hook-form';
import useSupport from './useSupport';
import {
  ContactFormInputProps,
  withContactFormSchema,
  withContactFormSchemaValues,
} from './validation';

export interface ContactFormProps {
  onSubmit?(values: ContactFormInputProps): void;
  url?: string;
}

const ContactForm: FC<ContactFormProps> = (props) => {
  const { onSubmit, url } = props;
  const { postSupport, hasError, isSuccess, isLoading } = useSupport();
  const { register, handleSubmit, formState, reset } =
    useForm<withContactFormSchemaValues>({
      resolver: yupResolver(withContactFormSchema),
      defaultValues: {
        emailAddress: '',
        message: '',
        details: {
          name: '',
        },
      },
      shouldUnregister: true,
    });

  const onSubmitForm = async (values: withContactFormSchemaValues) => {
    if (url) {
      await postSupport(url, values);

      reset({
        emailAddress: '',
        message: '',
        details: {
          name: '',
        },
      });
    }

    if (onSubmit) onSubmit(values);
  };

  const { isSubmitting, errors } = formState;

  return (
    <Box maxW={512} data-testid="box.contactform.container">
      <Box
        as="form"
        onSubmitCapture={handleSubmit(onSubmitForm)}
        data-testid="box.contactform.form"
      >
        <Stack spacing={4}>
          <TextField
            {...register('details.name')}
            id="name"
            label="Name"
            placeholder="Input your name"
            error={errors.details?.name?.message}
            isDisabled={isSubmitting}
          />
          <TextField
            {...register('emailAddress')}
            id="emailAddress"
            label="Email"
            placeholder="Input your email address"
            error={errors.emailAddress?.message}
            isDisabled={isSubmitting}
            data-testid="input.contactform.email"
          />
          <MultilineField
            {...register('message')}
            id="message"
            label="Desciption of concern"
            placeholder="Enter description"
            error={errors.message?.message}
            isDisabled={isSubmitting}
            data-testid="textarea.contactform.description"
          />
          <Button
            w="full"
            variant="primary"
            isLoading={isSubmitting || isLoading}
            type="submit"
            data-testid="button.contactform.submit"
          >
            Send
          </Button>

          {hasError && (
            <Alert status="error">
              <AlertIcon />
              <AlertDescription>Oops, Something went wrong.</AlertDescription>
            </Alert>
          )}

          {isSuccess && (
            <Alert status="success">
              <AlertIcon />
              <AlertDescription>Message successfully sent!</AlertDescription>
            </Alert>
          )}
        </Stack>
      </Box>
    </Box>
  );
};

export default ContactForm;
