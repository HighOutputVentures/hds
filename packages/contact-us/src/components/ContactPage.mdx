import { Canvas, Meta, Story, ArgsTable } from '@storybook/addon-docs';

<Meta title="Layouts/Contact/Contact Page" />

# HDS Contact Us

To install package, run command:

```bash
npm i @highoutput/hds-contact-us
```

<Canvas>
  <Story id="layouts-contact-contact-page--default"></Story>
</Canvas>

<ArgsTable />
