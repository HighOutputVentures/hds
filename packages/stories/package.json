{
  "name": "stories",
  "version": "1.0.0-next",
  "main": "./index.js",
  "types": "./index.d.ts",
  "exports": {
    ".": {
      "import": "./index.mjs",
      "require": "./index.js"
    }
  }
}
