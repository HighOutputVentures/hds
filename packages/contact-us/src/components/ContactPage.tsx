import { Box, Center, Text } from '@chakra-ui/react';
import React, { FC, ReactNode } from 'react';
import { HovIcon } from '@highoutput/hds-icons';

import ContactCard from './ContactCard';

// type WithoutChildren<T> = Omit<T, 'children'>;

export interface ContagePageProps {
  title?: string;
  secondaryTitle?: string;
  iconNode?: ReactNode;
  url?: string;
  // partProps?: {
  //   contactCard?: WithoutChildren<ContactCardProps>;
  //   title?: WithoutChildren<TextProps>;
  //   secondaryTitle?: WithoutChildren<TextProps>;
  // };
}

const ContactPage: FC<ContagePageProps> = (props) => {
  const {
    title = 'Contact Us',
    secondaryTitle = `Tell us what you need and we'll help you out.`,
    iconNode,
    url,
  } = props;
  return (
    <Box pos="relative" w="100%" data-testid="box.contactpage.container">
      <Box
        minW={'600px'}
        bg="#000"
        w="100%"
        h="400px"
        // {...props}
        data-testid="box.contactpage.banner"
      >
        <Center pt={20} pb={8} data-testid="center.contactpage.iconposition">
          {iconNode ? (
            iconNode
          ) : (
            <HovIcon color="white" data-testid="icon.contactpage.hovicon" />
          )}
        </Center>

        <Center data-testid="center.contactpage.titlepostion">
          <Text
            size="heading-web-3"
            color="white"
            // {...partProps?.title}
            data-testid="text.contactpage.title"
          >
            {title}
          </Text>
        </Center>

        <Center data-testid="center.contactpage.secondarytitleposition">
          <Text
            color="gray.200"
            // {...partProps?.secondaryTitle}
            data-testid="text.contactpage.secondarytitle"
          >
            {secondaryTitle}
          </Text>
        </Center>
      </Box>
      <Center
        minW={'600px'}
        data-testid="center.contactpage.secondarytitleposition"
      >
        <ContactCard
          pos="absolute"
          top="250px"
          w="512px"
          url={url}
          // {...partProps?.contactCard}
        />
      </Center>
    </Box>
  );
};

export default ContactPage;
