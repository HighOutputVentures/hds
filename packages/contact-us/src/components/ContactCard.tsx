import { Box, BoxProps, Center, Text } from '@highoutput/hds';
import React, { FC, ReactNode } from 'react';
import ContactForm from './ContactForm';

export interface ContactCardProps extends Pick<BoxProps, 'pos' | 'top' | 'w'> {
  children?: ReactNode;
  title?: string;
  url?: string;
}

const ContactCard: FC<ContactCardProps> = (props) => {
  const { children, title = 'Drop your message', url } = props;
  return (
    <Box
      w={512}
      bg="white"
      borderRadius="8px"
      padding="56px"
      boxShadow="0px 10px 15px -3px rgba(0, 0, 0, 0.1), 0px 4px 6px -2px rgba(0, 0, 0, 0.05)"
      {...props}
      data-testid="box.contactcard.container"
    >
      <Center mb={8} data-testid="center.contactcard.titleposition">
        <Text size="header-4" data-testid="text.contactcard.title">
          {title}
        </Text>
      </Center>
      {children ? (
        children
      ) : (
        <ContactForm
          // {...partProps?.contactForm}
          url={url}
        />
      )}
    </Box>
  );
};

export default ContactCard;
