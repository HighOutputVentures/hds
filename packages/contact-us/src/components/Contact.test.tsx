import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';

import renderer from 'react-test-renderer';
import ContactCard from './ContactCard';
import { Default as ContactCardComponent } from './ContactCard.stories';
import ContactForm from './ContactForm';
import { Default as ContactFormComponent } from './ContactForm.stories';
import ContactPage from './ContactPage';
import { Default as ContactPageComponent } from './ContactPage.stories';

jest.mock('react-textarea-autosize', () => {
  return {
    __esModule: true,
    default({ minRows: _mxr, maxRows: _mnr, ...props }: any) {
      return <textarea {...props} />;
    },
  };
});

describe.skip('Contact Form Component', () => {
  beforeEach(() => {
    render(<ContactForm />);
  });

  it('should render contact form container', async () => {
    const contactFormContainer = await screen.findAllByTestId(
      'box.contactform.container'
    );
    expect(contactFormContainer).toHaveLength(1);
  });

  it('should render contact form as form', async () => {
    const contactFormAsForm = await screen.findAllByTestId(
      'box.contactform.form'
    );
    expect(contactFormAsForm).toHaveLength(1);
  });

  it('should render 2 input field', async () => {
    const inputFields = await screen.findAllByTestId(/hds.input.field/i);
    expect(inputFields).toHaveLength(2);
  });

  it('should render input field name', async () => {
    const { container } = render(<ContactForm />);
    const inputName = container.querySelector(`input[id="name"]`);
    expect(inputName).toBeInTheDocument();
  });

  it('should render input field email', async () => {
    const { container } = render(<ContactForm />);
    const inputEmail = container.querySelector(`input[id="emailAddress"]`);
    expect(inputEmail).toBeInTheDocument();
  });

  it('should render textarea field description', async () => {
    const { container } = render(<ContactForm />);
    const textareaDescription = container.querySelector(
      `textarea[id="message"]`
    );
    expect(textareaDescription).toBeInTheDocument();
  });

  it('should render button submit', async () => {
    const submitButton = await screen.findAllByTestId(
      'button.contactform.submit'
    );
    expect(submitButton).toHaveLength(1);
  });

  test('user clicks submit in contact form with no value and renders error messages', async () => {
    const submit = await screen.findByTestId('box.contactform.form');
    await fireEvent.submit(submit);
    const errorFormContorl = await screen.findAllByTestId(
      /hds.form.control.error.message/i
    );
    expect(errorFormContorl).toHaveLength(3);
  });

  test('user clicks submit in contact form must contain name is required', async () => {
    const submit = await screen.findByTestId('box.contactform.form');
    await fireEvent.submit(submit);
    const nameRequired = await screen.findByText(/name is required./i);
    expect(nameRequired).toBeInTheDocument();
  });

  test('user clicks submit in contact form must contain email is required', async () => {
    const submit = await screen.findByTestId('box.contactform.form');
    await fireEvent.submit(submit);
    const nameRequired = await screen.findByText(/email is required./i);
    expect(nameRequired).toBeInTheDocument();
  });

  test('user clicks submit in contact form must contain description is required', async () => {
    const submit = await screen.findByTestId('box.contactform.form');
    await fireEvent.submit(submit);
    const nameRequired = await screen.findByText(/description is required./i);
    expect(nameRequired).toBeInTheDocument();
  });

  test('contact form email validation', async () => {
    const emailInput = await screen.findByPlaceholderText(
      'Input your email address'
    );
    await fireEvent.change(emailInput, { target: { value: 'test' } });
    const submit = await screen.findByTestId('box.contactform.form');
    await fireEvent.submit(submit);
    const emailValid = await screen.findByText(
      /please enter a valid email address./i
    );
    expect(emailValid).toBeInTheDocument();
  });

  describe('Snapshot', () => {
    it('(ContactForm) Should match snapshot', () => {
      const component = renderer.create(<ContactForm />);

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
      component.unmount();
    });
  });
});

describe.skip('Storybook - Contact Form Component', () => {
  beforeEach(() => {
    render(<ContactFormComponent />);
  });

  it('should render contact form container', async () => {
    const contactFormContainer = await screen.findAllByTestId(
      'box.contactform.container'
    );
    expect(contactFormContainer).toHaveLength(1);
  });

  it('should render contact form as form', async () => {
    const contactFormAsForm = await screen.findAllByTestId(
      'box.contactform.form'
    );
    expect(contactFormAsForm).toHaveLength(1);
  });

  it('should render 2 input field', async () => {
    const inputFields = await screen.findAllByTestId(/hds.input.field/i);
    expect(inputFields).toHaveLength(2);
  });

  it('should render input field name', async () => {
    const { container } = render(<ContactForm />);
    const inputName = container.querySelector(`input[id="name"]`);
    expect(inputName).toBeInTheDocument();
  });

  it('should render input field email', async () => {
    const { container } = render(<ContactForm />);
    const inputEmail = container.querySelector(`input[id="emailAddress"]`);
    expect(inputEmail).toBeInTheDocument();
  });

  it('should render textarea field description', async () => {
    const { container } = render(<ContactForm />);
    const textareaDescription = container.querySelector(
      `textarea[id="message"]`
    );
    expect(textareaDescription).toBeInTheDocument();
  });

  it('should render button submit', async () => {
    const submitButton = await screen.findAllByTestId(
      'button.contactform.submit'
    );
    expect(submitButton).toHaveLength(1);
  });

  test('user clicks submit in contact form with no value and renders error messages', async () => {
    const submit = await screen.findByTestId('box.contactform.form');
    await fireEvent.submit(submit);
    const errorFormContorl = await screen.findAllByTestId(
      /hds.form.control.error.message/i
    );
    expect(errorFormContorl).toHaveLength(3);
  });

  test('user clicks submit in contact form must contain name is required', async () => {
    const submit = await screen.findByTestId('box.contactform.form');
    await fireEvent.submit(submit);
    const nameRequired = await screen.findByText(/name is required./i);
    expect(nameRequired).toBeInTheDocument();
  });

  test('user clicks submit in contact form must contain email is required', async () => {
    const submit = await screen.findByTestId('box.contactform.form');
    await fireEvent.submit(submit);
    const nameRequired = await screen.findByText(/email is required./i);
    expect(nameRequired).toBeInTheDocument();
  });

  test('user clicks submit in contact form must contain description is required', async () => {
    const submit = await screen.findByTestId('box.contactform.form');
    await fireEvent.submit(submit);
    const nameRequired = await screen.findByText(/description is required./i);
    expect(nameRequired).toBeInTheDocument();
  });

  test('contact form email validation', async () => {
    const emailInput = await screen.findByPlaceholderText(
      'Input your email address'
    );
    await fireEvent.change(emailInput, { target: { value: 'test' } });
    const submit = await screen.findByTestId('box.contactform.form');
    await fireEvent.submit(submit);
    const emailValid = await screen.findByText(
      /please enter a valid email address./i
    );
    expect(emailValid).toBeInTheDocument();
  });
});

describe.skip('Contact Card Component', () => {
  beforeEach(() => {
    render(<ContactCard />);
  });

  it('should render contact card container', async () => {
    const contactCardContainer = await screen.findAllByTestId(
      'box.contactcard.container'
    );
    expect(contactCardContainer).toHaveLength(1);
  });

  it('should render title in center position', async () => {
    const titleCenter = await screen.findAllByTestId(
      'center.contactcard.titleposition'
    );
    expect(titleCenter).toHaveLength(1);
  });

  it('should render title', async () => {
    const title = await screen.findAllByTestId('text.contactcard.title');
    expect(title).toHaveLength(1);
  });

  it('should render title pass from props', async () => {
    const { getByText } = render(<ContactCard title="Test title" />);
    const title = getByText(/test title/i);
    expect(title).toBeInTheDocument();
  });

  describe('Snapshot', () => {
    it('(ContactCard) Should match snapshot', () => {
      const component = renderer.create(<ContactCard />);

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
      component.unmount();
    });
  });
});

describe.skip('Storybook - Contact Card Component', () => {
  beforeEach(() => {
    render(<ContactCardComponent />);
  });

  it('should render contact card container', async () => {
    const contactCardContainer = await screen.findAllByTestId(
      'box.contactcard.container'
    );
    expect(contactCardContainer).toHaveLength(1);
  });

  it('should render title in center position', async () => {
    const titleCenter = await screen.findAllByTestId(
      'center.contactcard.titleposition'
    );
    expect(titleCenter).toHaveLength(1);
  });

  it('should render title', async () => {
    const title = await screen.findAllByTestId('text.contactcard.title');
    expect(title).toHaveLength(1);
  });

  it('should render title pass from props', async () => {
    const { getByText } = render(<ContactCardComponent title="Test title" />);
    const title = getByText(/test title/i);
    expect(title).toBeInTheDocument();
  });
});

describe.skip('Contact Page Component', () => {
  beforeEach(() => {
    render(<ContactPage />);
  });

  it('should render contact page container', async () => {
    const contactPageContainer = await screen.findAllByTestId(
      'box.contactpage.container'
    );
    expect(contactPageContainer).toHaveLength(1);
  });

  it('should render contact page banner', async () => {
    const contactPageBanner = await screen.findAllByTestId(
      'box.contactpage.banner'
    );
    expect(contactPageBanner).toHaveLength(1);
  });

  it('should render hov icon in center', async () => {
    const hovIconPosition = await screen.findAllByTestId(
      'center.contactpage.iconposition'
    );
    expect(hovIconPosition).toHaveLength(1);
  });

  it('should render hov icon', async () => {
    const hovIcon = await screen.findAllByTestId('icon.contactpage.hovicon');
    expect(hovIcon).toHaveLength(1);
  });

  it('should render title center position', async () => {
    const titleCenterPosition = await screen.findAllByTestId(
      'center.contactpage.titlepostion'
    );
    expect(titleCenterPosition).toHaveLength(1);
  });

  it('should render title', async () => {
    const title = await screen.findAllByTestId('text.contactpage.title');
    expect(title).toHaveLength(1);
  });

  describe('Snapshot', () => {
    it('(ContactPage) Should match snapshot', () => {
      const component = renderer.create(<ContactPage />);

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
      component.unmount();
    });
  });
});

describe.skip('Storybook - Contact Page Component', () => {
  beforeEach(() => {
    render(<ContactPageComponent />);
  });

  it('should render contact page container', async () => {
    const contactPageContainer = await screen.findAllByTestId(
      'box.contactpage.container'
    );
    expect(contactPageContainer).toHaveLength(1);
  });

  it('should render contact page banner', async () => {
    const contactPageBanner = await screen.findAllByTestId(
      'box.contactpage.banner'
    );
    expect(contactPageBanner).toHaveLength(1);
  });

  it('should render hov icon in center', async () => {
    const hovIconPosition = await screen.findAllByTestId(
      'center.contactpage.iconposition'
    );
    expect(hovIconPosition).toHaveLength(1);
  });

  it('should render hov icon', async () => {
    const hovIcon = await screen.findAllByTestId('icon.contactpage.hovicon');
    expect(hovIcon).toHaveLength(1);
  });

  it('should render title center position', async () => {
    const titleCenterPosition = await screen.findAllByTestId(
      'center.contactpage.titlepostion'
    );
    expect(titleCenterPosition).toHaveLength(1);
  });

  it('should render title', async () => {
    const title = await screen.findAllByTestId('text.contactpage.title');
    expect(title).toHaveLength(1);
  });
});
