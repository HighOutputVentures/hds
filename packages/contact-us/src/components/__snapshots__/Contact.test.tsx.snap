// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Card Component Snapshot (ContactCard) Should match snapshot 1`] = `
<div
  className="css-xiku2a"
  data-testid="box.contactcard.container"
>
  <div
    className="css-er5rnb"
    data-testid="center.contactcard.titleposition"
  >
    <p
      className="chakra-text css-0"
      data-testid="text.contactcard.title"
    >
      Drop your message
    </p>
  </div>
  <div
    className="css-1kck0el"
    data-testid="box.contactform.container"
  >
    <form
      className="css-0"
      data-testid="box.contactform.form"
      onSubmitCapture={[Function]}
    >
      <div
        className="chakra-stack css-17kdbol"
      >
        <div
          aria-label="Form Group"
          className="chakra-form-control css-0"
          data-testid=":ra:-form-container-form-control"
          role="group"
        >
          <label
            className="chakra-form__label css-168uyi5"
            data-testid=":ra:-form-container-label"
            htmlFor="name"
            id="name-label"
          >
            Name
          </label>
          <div
            className="chakra-input__group css-4302v8"
            data-testid=":r9:-input-field-group"
          >
            <input
              className="chakra-input css-0"
              data-testid=":r9:-input-field-input"
              disabled={false}
              id="name"
              name="details.name"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="Input your name"
              readOnly={false}
              required={false}
              role="input"
              type="text"
            />
          </div>
        </div>
        <div
          aria-label="Form Group"
          className="chakra-form-control css-0"
          data-testid=":rd:-form-container-form-control"
          role="group"
        >
          <label
            className="chakra-form__label css-168uyi5"
            data-testid=":rd:-form-container-label"
            htmlFor="emailAddress"
            id="emailAddress-label"
          >
            Email
          </label>
          <div
            className="chakra-input__group css-4302v8"
            data-testid=":rc:-input-field-group"
          >
            <input
              className="chakra-input css-0"
              data-testid=":rc:-input-field-input"
              disabled={false}
              id="emailAddress"
              name="emailAddress"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="Input your email address"
              readOnly={false}
              required={false}
              role="input"
              type="text"
            />
          </div>
        </div>
        <div
          aria-label="Form Group"
          className="chakra-form-control css-0"
          data-testid=":rg:-form-container-form-control"
          role="group"
        >
          <label
            className="chakra-form__label css-168uyi5"
            data-testid=":rg:-form-container-label"
            htmlFor="message"
            id="message-label"
          >
            Desciption of concern
          </label>
          <div
            className="chakra-input__group css-4302v8"
          >
            <textarea
              className="chakra-textarea css-djz069"
              data-testid=":rf:-textarea-field-input"
              disabled={false}
              id="message"
              label="Desciption of concern"
              name="message"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Enter description"
              readOnly={false}
              required={false}
            />
          </div>
        </div>
        <button
          className="chakra-button css-l01cow"
          data-testid="button.contactform.submit"
          disabled={false}
          type="submit"
        >
          Send
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`Contact Form Component Snapshot (ContactForm) Should match snapshot 1`] = `
<div
  className="css-1kck0el"
  data-testid="box.contactform.container"
>
  <form
    className="css-0"
    data-testid="box.contactform.form"
    onSubmitCapture={[Function]}
  >
    <div
      className="chakra-stack css-17kdbol"
    >
      <div
        aria-label="Form Group"
        className="chakra-form-control css-0"
        data-testid=":r1:-form-container-form-control"
        role="group"
      >
        <label
          className="chakra-form__label css-168uyi5"
          data-testid=":r1:-form-container-label"
          htmlFor="name"
          id="name-label"
        >
          Name
        </label>
        <div
          className="chakra-input__group css-4302v8"
          data-testid=":r0:-input-field-group"
        >
          <input
            className="chakra-input css-0"
            data-testid=":r0:-input-field-input"
            disabled={false}
            id="name"
            name="details.name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            placeholder="Input your name"
            readOnly={false}
            required={false}
            role="input"
            type="text"
          />
        </div>
      </div>
      <div
        aria-label="Form Group"
        className="chakra-form-control css-0"
        data-testid=":r4:-form-container-form-control"
        role="group"
      >
        <label
          className="chakra-form__label css-168uyi5"
          data-testid=":r4:-form-container-label"
          htmlFor="emailAddress"
          id="emailAddress-label"
        >
          Email
        </label>
        <div
          className="chakra-input__group css-4302v8"
          data-testid=":r3:-input-field-group"
        >
          <input
            className="chakra-input css-0"
            data-testid=":r3:-input-field-input"
            disabled={false}
            id="emailAddress"
            name="emailAddress"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            placeholder="Input your email address"
            readOnly={false}
            required={false}
            role="input"
            type="text"
          />
        </div>
      </div>
      <div
        aria-label="Form Group"
        className="chakra-form-control css-0"
        data-testid=":r7:-form-container-form-control"
        role="group"
      >
        <label
          className="chakra-form__label css-168uyi5"
          data-testid=":r7:-form-container-label"
          htmlFor="message"
          id="message-label"
        >
          Desciption of concern
        </label>
        <div
          className="chakra-input__group css-4302v8"
        >
          <textarea
            className="chakra-textarea css-djz069"
            data-testid=":r6:-textarea-field-input"
            disabled={false}
            id="message"
            label="Desciption of concern"
            name="message"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter description"
            readOnly={false}
            required={false}
          />
        </div>
      </div>
      <button
        className="chakra-button css-l01cow"
        data-testid="button.contactform.submit"
        disabled={false}
        type="submit"
      >
        Send
      </button>
    </div>
  </form>
</div>
`;

exports[`Contact Page Component Snapshot (ContactPage) Should match snapshot 1`] = `
<div
  className="css-1jke4yk"
  data-testid="box.contactpage.container"
>
  <div
    className="css-1yom1oh"
    data-testid="box.contactpage.banner"
  >
    <div
      className="css-1eqrob5"
      data-testid="center.contactpage.iconposition"
    >
      <svg
        className="chakra-icon css-1rug128"
        data-testid="icon.contactpage.hovicon"
        focusable={false}
        viewBox="0 0 32 32"
      >
        <path
          d="M0 32L3.55556 28.4444V32H0ZM14.2222 32V26.6667L10.6667 30.2222V32H7.11111V24.8889L10.6667 21.3333V24.8889V26.6667L14.2222 23.1111V17.7778L17.7778 14.2222V17.7778V24.8889V32H14.2222ZM28.4444 17.7778V10.6667V7.11111L24.8889 10.6667V17.7778V28.4444H28.4444V17.7778ZM28.4444 32H24.8889H21.3333V17.7778V10.6667L24.8889 7.11111L28.4444 3.55556L32 0V3.55556V10.6667V17.7778V32H28.4444Z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      className="css-gmuwbf"
      data-testid="center.contactpage.titlepostion"
    >
      <p
        className="chakra-text css-kbn7if"
        data-testid="text.contactpage.title"
      >
        Contact Us
      </p>
    </div>
    <div
      className="css-gmuwbf"
      data-testid="center.contactpage.secondarytitleposition"
    >
      <p
        className="chakra-text css-gnzpcw"
        data-testid="text.contactpage.secondarytitle"
      >
        Tell us what you need and we'll help you out.
      </p>
    </div>
  </div>
  <div
    className="css-1iuukto"
    data-testid="center.contactpage.secondarytitleposition"
  >
    <div
      className="css-14kz8y"
      data-testid="box.contactcard.container"
    >
      <div
        className="css-er5rnb"
        data-testid="center.contactcard.titleposition"
      >
        <p
          className="chakra-text css-0"
          data-testid="text.contactcard.title"
        >
          Drop your message
        </p>
      </div>
      <div
        className="css-1kck0el"
        data-testid="box.contactform.container"
      >
        <form
          className="css-0"
          data-testid="box.contactform.form"
          onSubmitCapture={[Function]}
        >
          <div
            className="chakra-stack css-17kdbol"
          >
            <div
              aria-label="Form Group"
              className="chakra-form-control css-0"
              data-testid=":rj:-form-container-form-control"
              role="group"
            >
              <label
                className="chakra-form__label css-168uyi5"
                data-testid=":rj:-form-container-label"
                htmlFor="name"
                id="name-label"
              >
                Name
              </label>
              <div
                className="chakra-input__group css-4302v8"
                data-testid=":ri:-input-field-group"
              >
                <input
                  className="chakra-input css-0"
                  data-testid=":ri:-input-field-input"
                  disabled={false}
                  id="name"
                  name="details.name"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyPress={[Function]}
                  placeholder="Input your name"
                  readOnly={false}
                  required={false}
                  role="input"
                  type="text"
                />
              </div>
            </div>
            <div
              aria-label="Form Group"
              className="chakra-form-control css-0"
              data-testid=":rm:-form-container-form-control"
              role="group"
            >
              <label
                className="chakra-form__label css-168uyi5"
                data-testid=":rm:-form-container-label"
                htmlFor="emailAddress"
                id="emailAddress-label"
              >
                Email
              </label>
              <div
                className="chakra-input__group css-4302v8"
                data-testid=":rl:-input-field-group"
              >
                <input
                  className="chakra-input css-0"
                  data-testid=":rl:-input-field-input"
                  disabled={false}
                  id="emailAddress"
                  name="emailAddress"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyPress={[Function]}
                  placeholder="Input your email address"
                  readOnly={false}
                  required={false}
                  role="input"
                  type="text"
                />
              </div>
            </div>
            <div
              aria-label="Form Group"
              className="chakra-form-control css-0"
              data-testid=":rp:-form-container-form-control"
              role="group"
            >
              <label
                className="chakra-form__label css-168uyi5"
                data-testid=":rp:-form-container-label"
                htmlFor="message"
                id="message-label"
              >
                Desciption of concern
              </label>
              <div
                className="chakra-input__group css-4302v8"
              >
                <textarea
                  className="chakra-textarea css-djz069"
                  data-testid=":ro:-textarea-field-input"
                  disabled={false}
                  id="message"
                  label="Desciption of concern"
                  name="message"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Enter description"
                  readOnly={false}
                  required={false}
                />
              </div>
            </div>
            <button
              className="chakra-button css-l01cow"
              data-testid="button.contactform.submit"
              disabled={false}
              type="submit"
            >
              Send
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
